File: day:10/strStr.java
class Solution {
    public int strStr(String haystack, String needle) {
        // base case
        if(needle.length() == 0) return 0;
        if(haystack.equals(needle)) return 0;

        int sliceLength = needle.length();

        for(int i = 0; i < haystack.length(); i++) {
            int window = (i + sliceLength);
            if(window <= haystack.length()) {
                if(haystack.substring(i, window).equals(needle)) {
                    return i;
                }
            }
            else {
                return -1;
            }
        }
        return -1;
        }
}

File: day:10/strStr.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        # base case
        if(len(needle) == 0): return 0;
        if(haystack == needle): return 0;
    
        sliceLength = len(needle);
        
        for i in range(len(haystack)):
            window = (i + sliceLength)
            
            if window <= len(haystack):
                if(haystack[i:window] == needle):
                    return i
            else:
                return -1
        
        return -1
         
        
